registry/session.go

```diff
@@ -20,64 +21,96 @@ import (
 	"github.com/Sirupsen/logrus"
 	"github.com/docker/docker/cliconfig"
 	"github.com/docker/docker/pkg/httputils"
-	"github.com/docker/docker/pkg/requestdecorator"
 	"github.com/docker/docker/pkg/tarsum"
 )
 
 type Session struct {
-	authConfig    *cliconfig.AuthConfig
-	reqFactory    *requestdecorator.RequestFactory
 	indexEndpoint *Endpoint
-	jar           *cookiejar.Jar
-	timeout       TimeoutType
+	client        *http.Client
+	// TODO: remove authConfig
```

Comment:
`authConfig` definitely needs a home but I am not sure where it belongs. Seems like either on `Session` or `Service`. Typically, I would layer this as part of the `Session` but I am not sure if that is the right place for this package.